### PR TITLE
Support histogram metrics from engine in cache

### PR DESCRIPTION
**Pull request by Jeffwan**
## Pull Request Description
Support histogram metrics from engine in cache

## Related Issues
Resolves: phase I of #423 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: pkg/plugins/gateway/algorithms/throughput.go [hunk 1]
```diff
@@ -54,24 +54,24 @@ func (r throughputRouter) Route(ctx context.Context, pods map[string]*v1.Pod) (s
...
 		if err != nil {
 			klog.Error(err)
 			continue
 		}
 
 		// processing prompt tokens is twice as expensive than generation tokens
-		totalthroughput := 2*promptThroughput + generationThroughput
-		klog.V(4).Infof("pod: %v, podIP: %v, promptThroughput: %v, generationThroughput: %v, totalthroughput: %v",
-			pod.Name, pod.Status.PodIP, promptThroughput, generationThroughput, totalthroughput)
+		total_throughput := 2*promptThroughput.Value + generationThroughput.Value
```
Comment (kr11) [inline pkg/plugins/gateway/algorithms/throughput.go:69]: In autoscaler and the gateway, there exist two similar parts of metrics codes. Is this a good design? Shall we need to extract a set of common Metrics?

Comment (brosoul) [reply pkg/plugins/gateway/algorithms/throughput.go:69]: minor,   it is better to use `camel Case` naming (`totalThroughput`) for variable `total_throughput`

Comment (Jeffwan) [reply pkg/plugins/gateway/algorithms/throughput.go:69]: yeah, golang should use camel Case. my careless.

Comment (Jeffwan) [reply pkg/plugins/gateway/algorithms/throughput.go:69]: > Is this a good design? Shall we need to extract a set of common Metrics?

Good point. We should provide an unified solution for both metric consumers.

**Pull request merged**